### PR TITLE
Support renegotiation in client based on response ContentType

### LINKED ISSUE #25464: Protobuf tests are failing
Currently there are 2 tests that are failing with protobufs:
- [k8s.io] Pods should be submitted and removed [Conformance]
- [k8s.io] kubelet [k8s.io] Clean up pods on node kubelet should be able to delete 10 pods per node in 1m0s.

### PR DESCRIPTION
**Pull request by wojtek-t**
Fix #25464

### CODE REVIEW

**Comment by wojtek-t** [comment]
It seems that I'm handling incorrectly some plain text - will take a look

**Comment by wojtek-t** [comment]
It seems that I finally fixed it. This time tests should pass.

File: pkg/client/restclient/request.go [hunk 1]
```diff
@@ -910,12 +910,26 @@ func (r *Request) transformResponse(resp *http.Response, req *http.Request) Resu
 		return Result{err: errors.FromObject(status)}
 	}
 
-	// TODO: Check ContentType.
+	contentType := resp.Header.Get("Content-Type")
+	var decoder runtime.Decoder
+	if contentType == r.content.ContentType {
```
Comment (smarterclayton) [inline pkg/client/restclient/request.go]: I'm not sure this is safe in all cases - the string content-type may only be partially equivalent.  I think we have to do a slightly more invasive check.

At a minimum, in your else block you need to call `mime.Parse(contentType)` and pass both parts to renegotiate.

Comment (wojtek-t) [inline pkg/client/restclient/request.go]: You mean that I should have:

else {
  mediaType, params, err := mime.Parse(contentType)
  if err != nil {
    return Result{err: "error parsing content type"}
  }
  decoder, err := r.serializers.RenegotiatedDecoder(mediaType, params)
  if err != nil {
    // return what I have
  }
}

or sth else?

Comment (smarterclayton) [inline pkg/client/restclient/request.go]: Basically yes.

**Comment by smarterclayton** [comment]
No other comments, self apply after fixing (if I'm not around).

File: pkg/client/restclient/request.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/client/restclient/request.go]: Done

**Comment by wojtek-t** [comment]
Comment addressed - self-applying lgtm label based on the above comment.
Thanks @smarterclayton !

**Comment by wojtek-t** [comment]
All tests has just passed. I'm merging it manually, because broken merge-bot is a flake.

**Comment by smarterclayton** [comment]
Retroactive LGTM as per my original comment ;)

**Comment by wojtek-t** [comment]
Ref #8132

**Pull request merged**